modules/login/php/requestaccount.class.inc

```diff
@@ -269,7 +269,67 @@ class RequestAccount extends \NDB_Form
             }
         }
 
+        // Verify reCAPTCHA
+        $err = [];
+        if (isset($values['g-recaptcha-response']) && isset($reCAPTCHAPrivate)) {
+            $recaptcha = new \ReCaptcha\ReCaptcha($reCAPTCHAPrivate);
+            $resp      = $recaptcha->verify(
+                $values['g-recaptcha-response'],
+                $_SERVER['REMOTE_ADDR']
+            );
+            if (!$resp->isSuccess()) {
+                $err['captcha'] = 'Please complete the reCaptcha!';
+            }
+        }
         // Show success message even if email already exists for security reasons
         $this->tpl_data['success'] = true;
+
+        // If there are error messages, return early without sending emails.
+        if (!empty($err)) {
+            return;
+        }
+
+        // Send an email to all users meeting the following criteria:
+        //     - Associated with the Site
+        //     - Associated with the Project
+        //     - Have the User Management permission
+        $query  = <<<SQL
+        SELECT DISTINCT u.Email,u.userID 
+        FROM users u 
+        JOIN user_psc_rel upr 
+        ON u.ID = upr.UserID 
+        JOIN user_project_rel uprojr 
+        ON u.ID = uprojr.UserID 
+        WHERE upr.CenterID=:center
+        AND uprojr.ProjectID=:project
+SQL;
+        $result = $DB->pselect(
+            $query,
+            ['center' => $site, 'project' => $project]
+        );
+        foreach ($result as $row) {
+            $hasPermission = \User::factory($row['userID'])
+                ->hasAnyPermission(
+                    ['user_accounts', 'user_accounts_multisite']
+                );
```

Comment:
I find this is a weird combination of permission checking. you are doing SQL checks for site and project and User object checks for the module permissions. could you justify ? why not all SQL or all user class?

Reply:
I would do all SQL but I'm not skilled enough with it to make a big query that does all of it. I'm also working everything out before my last day so I'm using the quickest approach to getting all requested changes made + everything rebased. Feel free to refactor it or create a ticket to clean this up but it's not something I'm going to be able to prioritize